masonry/src/render_root.rs

```diff
@@ -51,45 +51,92 @@ const INVALID_IME_AREA: Rect = Rect::new(f64::NAN, f64::NAN, f64::NAN, f64::NAN)
 ///
 /// This is also the type that owns the widget tree.
 pub struct RenderRoot {
+    /// Root of the widget tree.
     pub(crate) root: WidgetPod<Box<dyn Widget>>,
+
+    /// Whether the window size should be determined by the content or the user.
     pub(crate) size_policy: WindowSizePolicy,
+
+    /// Current size of the window.
     pub(crate) size: PhysicalSize<u32>,
-    // TODO - Currently this is always 1.0
-    // kurbo coordinates are assumed to be in logical pixels
+
+    /// DPI scale factor.
+    ///
+    /// Kurbo coordinates are assumed to be in logical pixels
     pub(crate) scale_factor: f64,
+
     /// Is `Some` if the most recently displayed frame was an animation frame.
     pub(crate) last_anim: Option<Instant>,
+
+    /// Last mouse position. Updated by `on_pointer_event` pass, used by other passes.
     pub(crate) last_mouse_pos: Option<LogicalPosition<f64>>,
-    pub(crate) cursor_icon: CursorIcon,
+
+    /// State passed to context types.
     pub(crate) global_state: RenderRootState,
-    // TODO - Add "access_tree_active" to detect when you don't need to update the
+
+    /// Whether the next accessibility pass should rebuild the entire access tree.
+    ///
+    /// TODO - Add `access_tree_active` to detect when you don't need to update the
     // access tree
     pub(crate) rebuild_access_tree: bool,
+
+    /// The widget tree; stores widgets and their states.
     pub(crate) widget_arena: WidgetArena,
 }
 
-// TODO - Document these fields.
+/// State shared between passes.
 pub(crate) struct RenderRootState {
+    /// Queue of signals to be processed by the event loop.
     pub(crate) signal_queue: VecDeque<RenderRootSignal>,
+
+    /// Currently focused widget.
     pub(crate) focused_widget: Option<WidgetId>,
+
+    /// List of ancestors of the currently focused widget.
     pub(crate) focused_path: Vec<WidgetId>,
+
+    /// Widget that will be focused once the `update_focus` pass is run.
     pub(crate) next_focused_widget: Option<WidgetId>,
+
+    /// Most recently clicked widget.
+    ///
+    /// This is used to pick the focused widget on Tab events.
     pub(crate) most_recently_clicked_widget: Option<WidgetId>,
+
+    /// Widgets that have requested to be scrolled into view.
     pub(crate) scroll_request_targets: Vec<(WidgetId, Rect)>,
+
+    /// List of ancestors of the currently hovered widget.
```

Comment:
Maybe here would be a good reason to explain why we store this (i.e. so that hover state can be properly tracked after a reparenting)